src/eredis_client.erl

```diff
@@ -208,11 +208,8 @@ handle_info(Info, State) ->
 terminate(_Reason, State) ->
     case State#state.socket of
         undefined -> ok;
-        Socket    ->
-            case State#state.transport of
-                tls -> ssl:close(Socket);
-                _   -> gen_tcp:close(Socket)
-            end
+        Socket ->
+            close_transport(State, Socket)
```

Comment:
You could make close_transport accept an undefined socket and do nothing in that case. Then terminate is just one line.

Reply:
ah, good, just leaned that gen_server:terminate/2 ignores the return value anyway..